debian/compat

```diff
@@ -1 +1 @@
-10
+12
```

Comment:
I believe that, as of debhelper 11, the correct way to specify a compat level is to add a `Build-Depends: debhelper-compat (= 12)`.

https://manpages.debian.org/buster/debhelper/debhelper.7.en.html#COMPATIBILITY_LEVELS has some wods on this.

Reply:
Good point, will drop it.

Reply:
I keep the compat version on greater or equal to 12, because everything more recent than Debian buster has newer versions than that, so I think making this a hard equal would make this dependency unsatisfiable.

Reply:
No, this is incorrect.

again, see https://manpages.debian.org/buster/debhelper/debhelper.7.en.html#COMPATIBILITY_LEVELS.

You need a dependency on `debhelper-compat`, not `debhelper`. And it should be equal, not >=.

Reply:
Thinking some more about this: Is this a good idea? Since we want to keep this dependency satisfiable, the control file needs to have this `>= 12`, but without `debian/compat` saying which compatibility level to choose, won't this also choose higher compat levels when only a higher version debhelper is available in the repos? Since there are backwards incompatible changes between debhelper levels this sounds like a footgun.

Reply:
> You need a dependency on `debhelper-compat`, not `debhelper`. And it should be equal, not >=.

Ah, sorry, disregard the above thought then. I send it before reloading the page and glossed over the `-compat` in the docs, because of the stuff below it.